### PR TITLE
Removing nethcti-server local revision

### LINKED ISSUE #45: Pull nethcti-server from official repo branch
TODOs:
- [x] Check file sync between local copy and nethcti-server [ns8 branch](https://github.com/nethesis/nethcti-server/tree/ns8).
- [x] Update build process for cti-server, allowing pull from repository (example can be found in Nethesis/nethvoice#8)

### PR DESCRIPTION
**Pull request by Tbaile**
Closes #45

### CODE REVIEW

File: nethcti-server/Containerfile [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
-COPY usr/lib/node/nethcti-server/nethcti.js .
-COPY usr/local/bin/rpm /usr/local/bin/rpm
+# copy repository from alpine container
+COPY --from=repofetcher /app/root/var/lib/nethserver/nethcti/templates/customer_card /var/lib/nethserver/nethcti/templates/customer_card
+COPY --from=repofetcher /app/root/usr/lib/node/nethcti-server/plugins plugins
+COPY --from=repofetcher /app/root/usr/lib/node/nethcti-server/scripts scripts
+COPY --from=repofetcher /app/root/usr/lib/node/nethcti-server/sql sql
+COPY --from=repofetcher /app/root/usr/lib/node/nethcti-server/nethcti-cli.js .
+COPY --from=repofetcher /app/root/usr/lib/node/nethcti-server/nethcti.js .
+COPY --from=repofetcher /app/root/usr/local/bin/rpm /usr/local/bin/rpm
```
Comment (gsanchietti) [inline nethcti-server/Containerfile:35]: Preserve local copy for `rpm` command which is a fake bash script

File: nethcti-server/Containerfile [hunk 2]
```diff
@@ -1,28 +1,43 @@
+# clone repository using alpine container
+FROM docker.io/library/alpine:3.17.2 as repofetcher
+WORKDIR /app
+RUN apk add --no-cache git
+ARG NETHVOICE_SERVER_COMMIT=97ae4c9b054bdc289d660ea49da538c7862d2ac1
+RUN git clone https://github.com/nethesis/nethcti-server.git . \
+    && git checkout "${NETHVOICE_SERVER_COMMIT}"
```
Comment (gsanchietti) [inline nethcti-server/Containerfile:7]: I think this is the correct approach: you know exactly what's inside the image.
Still, I think in this development phase we are going to make many changes and we could forget to bump the hash.

What about, for now, just cloning the ns8 branch?
```suggestion
RUN git clone -b ns8 --depth=1 https://github.com/nethesis/nethcti-server.git .
```

Still, feel free to keep the current implementation.

**Review by gsanchietti — CHANGES_REQUESTED** [review]

File: nethcti-server/Containerfile (hunk 1 above)
Comment (Tbaile) [reply nethcti-server/Containerfile:35]: I've sent the file to `ns8` branch due to the similarity with `root/usr/lib/node/nethcti-server/scripts/ldap-authenticate.sh`.

I can totally put it here, no issues.

Comment (gsanchietti) [reply nethcti-server/Containerfile:35]: That's good.

File: nethcti-server/Containerfile (hunk 2 above)
Comment (gsanchietti) [reply nethcti-server/Containerfile:7]: Ok, let's keep it.
If we face development issues we will change it a bit.

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**